### PR TITLE
Allow use of API keys as alternative to account auth token for Twilio

### PR DESCRIPTION
**Pull request by mderynck**
Add live settings/env variables for:
TWILIO_API_KEY_SECRET
TWILIO_API_KEY_SID

This will allow more flexible configuration with OnCall to be configured using an issued Twilio API Key instead of only allowing Account SID/Auth Token.

### CODE REVIEW

**Comment by mderynck** [comment]
https://github.com/grafana/oncall/issues/136

File: engine/apps/base/models/live_setting.py [hunk 1]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
             "after you update them."
         ),
         "TWILIO_ACCOUNT_SID": (
-            "Twilio username to allow amixr send sms and make phone calls, "
+            "Twilio account SID/username to allow OnCall send sms and make phone calls, "
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:85]:
```suggestion
            "Twilio account SID/username to allow OnCall to send SMSes and make phone calls, "
```

File: engine/apps/base/models/live_setting.py [hunk 2]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
             "<a href='https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-How-to-Change-Them' target='_blank'>"
-            "more info</a>."
+            "more info</a>. Required."
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:87]:
```suggestion
            "here</a> for more info. Required."
```

File: engine/apps/base/models/live_setting.py [hunk 3]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
+        ),
+        "TWILIO_API_KEY_SID": (
+            "Twilio API key SID/username to allow OnCall send sms and make phone calls, "
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:90]:
```suggestion
            "Twilio API key SID/username to allow OnCall to send SMSes and make phone calls, see"
```

File: engine/apps/base/models/live_setting.py [hunk 4]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
+            "<a href='https://www.twilio.com/docs/iam/keys/api-key' target='_blank'>"
+            "more info</a>. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
+        ),
+        "TWILIO_API_KEY_SECRET": (
+            "Twilio API key secret/password to allow OnCall send sms and make phone calls, "
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:95]:
```suggestion
            "Twilio API key secret/password to allow OnCall to send SMSes and make phone calls, see"
```

File: engine/apps/base/models/live_setting.py [hunk 5]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
+            "<a href='https://www.twilio.com/docs/iam/keys/api-key' target='_blank'>"
+            "more info</a>. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:97]:
```suggestion
            "here</a> for more info. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
```

File: engine/apps/base/models/live_setting.py [hunk 6]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
         ),
         "TWILIO_AUTH_TOKEN": (
-            "Twilio password to allow amixr send sms and make calls, "
+            "Twilio password to allow OnCall send sms and make calls, "
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:100]:
```suggestion
            "Twilio password to allow OnCall to send SMSes and make calls, "
```

File: engine/apps/base/models/live_setting.py [hunk 7]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
             "<a href='https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-How-to-Change-Them' target='_blank'>"
-            "more info</a>."
+            "more info</a>. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:102]:
```suggestion
            "here</a> for more info. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
```

File: engine/apps/base/models/live_setting.py [hunk 8]
```diff
@@ -80,14 +82,24 @@ class LiveSetting(models.Model):
...
+            "here</a> for more info. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
         ),
         "TWILIO_AUTH_TOKEN": (
-            "Twilio password to allow amixr send sms and make calls, "
+            "Twilio password to allow OnCall to send SMSes and make calls, "
             "<a href='https://support.twilio.com/hc/en-us/articles/223136027-Auth-Tokens-and-How-to-Change-Them' target='_blank'>"
-            "more info</a>."
+            "here</a> for more info. Either (TWILIO_API_KEY_SID + TWILIO_API_KEY_SECRET) or TWILIO_AUTH_TOKEN is required."
         ),
         "TWILIO_NUMBER": (
```
Comment (joeyorlando) [inline engine/apps/base/models/live_setting.py:104]: I pushed [d0b8512](https://github.com/grafana/oncall/pull/643/commits/d0b851299267a8e1e091ee9205bfcb88fb222ec0) for some minor grammar changes.

Nitpick comment but maybe consider extracting some of the wording in `TWILIO_ACCOUNT_SID`, `TWILIO_API_KEY_SID`, `TWILIO_API_KEY_SECRET`, and `TWILIO_AUTH_TOKEN` into a variable.

**Review by joeyorlando — APPROVED** [review]
LGTM 👍 

Aside from the `CHANGELOG.md`, do we typically document changes like this somewhere in `/docs/sources`?

**Comment by mderynck** [comment]
@joeyorlando Thanks for looking at this, I found a few problems with my PR after more local testing, going to recreate after I work it out.

**Pull request closed without merging**